compiler/llvm/clangUtil.cpp

```diff
@@ -1222,6 +1222,40 @@ class CCodeGenConsumer : public ASTConsumer {
       adjustLayoutForGlobalToWide();
     }
 
+    void doHandleDecl(Decl* d) {
+      if (TypedefDecl *td = dyn_cast<TypedefDecl>(d)) {
+        const clang::Type *ctype= td->getUnderlyingType().getTypePtrOrNull();
+        if(ctype != NULL) {
+          info->lvt->addGlobalCDecl(td);
+        }
+      } else if (FunctionDecl *fd = dyn_cast<FunctionDecl>(d)) {
+        info->lvt->addGlobalCDecl(fd);
+      } else if (VarDecl *vd = dyn_cast<VarDecl>(d)) {
+        info->lvt->addGlobalCDecl(vd);
+      } else if (RecordDecl *rd = dyn_cast<RecordDecl>(d)) {
+        if( rd->getName().size() > 0 ) {
```

Comment:
why do we need this check?

Reply:
(just saw that you are just moving that from elsewhere)

Reply:
IIRC it's because there can be anonymous structs; in that case we don't want to try to save the declaration for later use under a name.